lib/js/components/Toast/Toast.vue

```diff
@@ -0,0 +1,286 @@
+	if (!el) {
+		return 0;
+	}
+
+	const style = window.getComputedStyle(el);
+	const margin = parseInt(style[`margin${direction}`], 10);
+	const padding = parseInt(style[`padding${direction}`], 10);
+	const border = parseInt(style[`border${direction}`], 10);
+
+	return margin + padding + border;
+}
+
+export default {
+	name: 'Toast',
+	components: {
+		DsButton,
+		DsCard,
+	},
+	props: {
+		size: {
+			type: String as PropType<ToastSizes>,
+			default: TOAST_SIZES.MEDIUM,
+		},
+		position: {
+			type: String as PropType<ToastPositions>,
+			default: TOAST_POSITIONS.BOTTOM_CENTER,
+		},
+		color: {
+			type: String as PropType<ToastColors>,
+			default: TOAST_COLORS.INFO,
+		},
+		footerPrimaryButtonText: {
+			type: String,
+			default: '',
+		},
+		footerPrimaryButtonIcon: {
+			type: Object,
+			default: null,
+			validator(footerPrimaryButtonIcon: any) {
+				return (
+					footerPrimaryButtonIcon == null ||
+					Object.values(ICONS).includes(toRaw(footerPrimaryButtonIcon))
+				);
+			},
+		},
+		footerSecondaryButtonText: {
+			type: String,
+			default: '',
+		},
+		footerSecondaryButtonIcon: {
+			type: Object,
+			default: null,
+			validator(footerSecondaryButtonIcon: any) {
+				return (
+					footerSecondaryButtonIcon == null ||
+					Object.values(ICONS).includes(toRaw(footerSecondaryButtonIcon))
+				);
+			},
+		},
+		isDisappearing: {
+			type: Boolean,
+			default: true,
+		},
+		disappearingTimeout: {
+			type: String,
+			default: '0',
+			validator(disappearingTimeout: string) {
+				return (
+					(disappearingTimeout === '0' || !isNaN(parseInt(disappearingTimeout, 10))) &&
+					parseInt(disappearingTimeout, 10) >= 0
+				);
+			},
+		},
+		appendTo: {
+			type: [String, HTMLElement],
+			default: null,
+		},
+	},
+	emits: {
+		close: () => true,
+		'primary-button-click': () => true,
+		'secondary-button-click': () => true,
+	},
+	data() {
+		return {
+			appendToElement: null,
+			BUTTON_COLORS: Object.freeze(BUTTON_COLORS),
+			BUTTON_RADIUSES: Object.freeze(BUTTON_RADIUSES),
+			BUTTON_TYPES: Object.freeze(BUTTON_TYPES),
+			TOAST_SIZES: Object.freeze(TOAST_SIZES),
+		};
+	},
+	computed: {
+		buttonPrimaryColor() {
+			return this.color === TOAST_COLORS.DANGER
+				? BUTTON_COLORS.NEUTRAL
+				: BUTTON_COLORS.PRIMARY;
+		},
+		buttonSecondaryColor() {
+			return this.color === TOAST_COLORS.DANGER
+				? BUTTON_COLORS.DANGER
+				: BUTTON_COLORS.NEUTRAL;
+		},
+		disableTeleport() {
+			return this.appendTo === null || this.appendToElement === this.$parent.$el;
+		},
+		styles() {
+			const parentWidthPx =
+				this.appendToElement?.offsetWidth - this.$refs.toastCard?.offsetWidth || 0;
+			const parentHeightPx =
+				this.appendToElement?.offsetHeight - this.$refs.toastCard?.offsetHeight || 0;
+			return {
+				bottomCenter: {
+					left: `${parentWidthPx / 2}px`,
+					top: `${Math.max(
+						parentHeightPx - TOAST_OFFSET - getOffset(this.appendToElement, 'Bottom'),
+						TOAST_OFFSET,
+					)}px`,
```

Comment:
why not bottom?
```suggestion
					bottom: `${TOAST_OFFSET - getOffset(this.appendToElement, 'Bottom')}px`,
```

Reply:
![Zrzut ekranu 2024-06-6 o 09 13 41](https://github.com/bethinkpl/design-system/assets/11304259/0af1e047-7528-4138-8393-c7719d436e3f)
Does not work